In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns 
from sklearn.cluster import KMeans
pd.options.display.max_columns = None
pd.options.display.max_rows = None

In [2]:
df = pd.read_csv('train_data.csv')

In [3]:
df.head()

index  lat       lon startdate  contest-pevpr-sfc-gauss-14d__pevpr  \
0      0  0.0  0.833333    9/1/14                              237.00   
1      1  0.0  0.833333    9/2/14                              228.90   
2      2  0.0  0.833333    9/3/14                              220.69   
3      3  0.0  0.833333    9/4/14                              225.28   
4      4  0.0  0.833333    9/5/14                              237.24   

   nmme0-tmp2m-34w__cancm30  nmme0-tmp2m-34w__cancm40  \
0                     29.02                     31.64   
1                     29.02                     31.64   
2                     29.02                     31.64   
3                     29.02                     31.64   
4                     29.02                     31.64   

   nmme0-tmp2m-34w__ccsm30  nmme0-tmp2m-34w__ccsm40  nmme0-tmp2m-34w__cfsv20  \
0                    29.57                    30.73                    29.71   
1                    29.57                    30.73                    29.71   
2                    29.57                    30.73                    29.71   
3                    29.57                    30.73                    29.71   
4                    29.57                    30.73                    29.71   

   nmme0-tmp2m-34w__gfdlflora0  nmme0-tmp2m-34w__gfdlflorb0  \
0                        31.52                        31.68   
1                        31.52                        31.68   
2                        31.52                        31.68   
3                        31.52                        31.68   
4                        31.52                        31.68   

   nmme0-tmp2m-34w__gfdl0  nmme0-tmp2m-34w__nasa0  nmme0-tmp2m-34w__nmme0mean  \
0                   30.56                   29.66                       30.46   
1                   30.56                   29.66                       30.46   
2                   30.56                   29.66                       30.46   
3                   30.56                   29.66                       30.46   
4                   30.56                   29.66                       30.46   

   contest-wind-h10-14d__wind-hgt-10  nmme-tmp2m-56w__cancm3  \
0                           31246.63                    28.3   
1                           31244.78                    28.3   
2                           31239.27                    28.3   
3                           31232.86                    28.3   
4                           31226.16                    28.3   

   nmme-tmp2m-56w__cancm4  nmme-tmp2m-56w__ccsm3  nmme-tmp2m-56w__ccsm4  \
0                   29.47                  27.13                  27.36   
1                   29.47                  27.13                  27.36   
2                   29.47                  27.13                  27.36   
3                   29.47                  27.13                  27.36   
4                   29.47                  27.13                  27.36   

   nmme-tmp2m-56w__cfsv2  nmme-tmp2m-56w__gfdl  nmme-tmp2m-56w__gfdlflora  \
0                  27.71                 28.25                       27.7   
1                  27.71                 28.25                       27.7   
2                  27.71                 28.25                       27.7   
3                  27.71                 28.25                       27.7   
4                  27.71                 28.25                       27.7   

   nmme-tmp2m-56w__gfdlflorb  nmme-tmp2m-56w__nasa  nmme-tmp2m-56w__nmmemean  \
0                      28.72                 28.38                     28.11   
1                      28.72                 28.38                     28.11   
2                      28.72                 28.38                     28.11   
3                      28.72                 28.38                     28.11   
4                      28.72                 28.38                     28.11   

   contest-rhum-sig995-14d__rhum  nmme-prate-34w__cancm3  \
0                          81.72  

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 375734 entries, 0 to 375733
Columns: 246 entries, index to wind-vwnd-925-2010-20
dtypes: float64(240), int64(4), object(2)
memory usage: 705.2+ MB


In [5]:
df.shape

(375734, 246)

### how is the mean calculated? where does the 3939 come from ? does that mean another nmme column has 3939 uniuqe values?

In [10]:
df.nmme0mean.nunique()

3939

In [13]:
df['nmme-tmp2m-34w__cancm4'].nunique()

4881

#### end of nmme nununique rabbit chase for now

In [6]:
df.describe().T

count           mean  \
index                                     375734.0  187866.500000   
lat                                       375734.0       0.592766   
lon                                       375734.0       0.517964   
contest-pevpr-sfc-gauss-14d__pevpr        375734.0     276.744064   
nmme0-tmp2m-34w__cancm30                  375734.0      10.801682   
nmme0-tmp2m-34w__cancm40                  375734.0      12.726739   
nmme0-tmp2m-34w__ccsm30                   359800.0      10.099323   
nmme0-tmp2m-34w__ccsm40                   375734.0      11.638229   
nmme0-tmp2m-34w__cfsv20                   375734.0      10.733886   
nmme0-tmp2m-34w__gfdlflora0               375734.0      11.772646   
nmme0-tmp2m-34w__gfdlflorb0               375734.0      11.694567   
nmme0-tmp2m-34w__gfdl0                    375734.0      11.037192   
nmme0-tmp2m-34w__nasa0                    375734.0      12.226311   
nmme0-tmp2m-34w__nmme0mean                375734.0      11.483127   
contest-wind-h10-14d__wind-hgt-10         375734.0   31071.557568   
nmme-tmp2m-56w__cancm3                    375734.0      10.376912   
nmme-tmp2m-56w__cancm4                    375734.0      12.651692   
nmme-tmp2m-56w__ccsm3                     365454.0       9.663224   
nmme-tmp2m-56w__ccsm4                     375734.0      11.931658   
nmme-tmp2m-56w__cfsv2                     375734.0      10.422056   
nmme-tmp2m-56w__gfdl                      375734.0      10.898743   
nmme-tmp2m-56w__gfdlflora                 375734.0      10.819885   
nmme-tmp2m-56w__gfdlflorb                 375734.0      10.793526   
nmme-tmp2m-56w__nasa                      375734.0      12.025167   
nmme-tmp2m-56w__nmmemean                  375734.0      11.087483   
contest-rhum-sig995-14d__rhum             375734.0      62.766512   
nmme-prate-34w__cancm3                    375734.0      24.773079   
nmme-prate-34w__cancm4                    375734.0      21.622735   
nmme-prate-34w__ccsm3                     366996.0      31.261547   
nmme-prate-34w__ccsm4                     375734.0      29.492811   
nmme-prate-34w__cfsv2                     375734.0      31.445859   
nmme-prate-34w__gfdl                      375734.0      28.085472   
nmme-prate-34w__gfdlflora                 375734.0      25.736902   
nmme-prate-34w__gfdlflorb                 375734.0      26.022777   
nmme-prate-34w__nasa                      375734.0      22.892876   
nmme-prate-34w__nmmemean                  375734.0      26.782739   
contest-wind-h100-14d__wind-hgt-100       375734.0   16404.561991   
nmme0-prate-56w__cancm30                  375734.0      24.280240   
nmme0-prate-56w__cancm40                  375734.0      20.243073   
nmme0-prate-56w__ccsm30                   359800.0      29.623993   
nmme0-prate-56w__ccsm40                   375734.0      28.562204   
nmme0-prate-56w__cfsv20                   375734.0      31.234376   
nmme0-prate-56w__gfdlflora0               375734.0      23.669086   
nmme0-prate-56w__gfdlflorb0               375734.0      23.871719   
nmme0-prate-56w__gfdl0                    375734.0      26.732080   
nmme0-prate-56w__nasa0                    375734.0      22.044573   
nmme0-prate-56w__nmme0mean                375734.0      25.526852   
nmme0-prate-34w__cancm30                  375734.0      24.250560   
nmme0-prate-34w__cancm40                  375734.0      20.253346   
nmme0-prate-34w__ccsm30                   359800.0      29.652606   
nmme0-prate-34w__ccsm40                   375734.0      28.516888   
nmme0-prate-34w__cfsv20                   375734.0      31.197654   
nmme0-prate-34w__gfdlflora0               375734.0      23.639439   
nmme0-prate-34w__gfdlflorb0               375734.0      23.834869   
nmme0-prate-34w__gfdl0                    375734.0      26.715076   
nmme0-prate-34w__nasa0                    375734.0      22.038805   
nmme0-prate-34w__nmme0mean                375734.0      25.508712   
contest-tmp2m-14d__tmp2m                  3

In [ ]:
df1.shape

### group by latitude to deternime how many locations.

In [15]:
df_lat = df.groupby('lat')

In [16]:
df_lat.lon.value_counts()

lat       lon     
0.000000  0.833333    731
          0.866667    731
0.045455  0.833333    731
          0.866667    731
          0.900000    731
0.090909  0.800000    731
          0.833333    731
          0.866667    731
          0.900000    731
          0.933333    731
0.136364  0.666667    731
          0.700000    731
          0.733333    731
          0.766667    731
          0.800000    731
          0.833333    731
          0.866667    731
          0.900000    731
          0.933333    731
          0.966667    731
          1.000000    731
0.181818  0.633333    731
          0.666667    731
          0.700000    731
          0.733333    731
          0.766667    731
          0.800000    731
          0.833333    731
          0.866667    731
          0.900000    731
          0.933333    731
          0.966667    731
          1.000000    731
0.227273  0.366667    731
          0.400000    731
          0.433333    731
          0.466667    731
          0.500000 

<div class="alert alert-block alert-success">
<b>Finding:</b>  375734 obervations with 731 per location means there are 514 locations use green boxes for 
related links from each section of a notebook.
</div>

In [ ]:
df_lat.head()

In [ ]:
df.lon.value_counts()
